### PR TITLE
resource/aws_wafv2_logging_configuration: add support for `logging_filter`

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes https://github.com/hashicorp/terraform-provider-aws/issues/18665

Output from acceptance testing:

```
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_emptyRedactedFields (373.82s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_basic (456.06s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_disappears (572.65s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_disappears_WebAcl (584.69s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateMethodRedactedField (585.70s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateEmptyRedactedFields (653.53s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeResourceARNForceNew (717.29s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateQueryStringRedactedField (738.86s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateUriPathRedactedField (738.91s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeLogDestinationConfigsForceNew (762.54s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateMultipleRedactedFields (827.28s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateSingleHeaderRedactedField (841.12s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_loggingFilter (865.88s)
```

### CODE REVIEW

File: aws/resource_aws_wafv2_web_acl_logging_configuration.go [hunk 1]
```diff
@@ -37,6 +38,78 @@ func resourceAwsWafv2WebACLLoggingConfiguration() *schema.Resource {
...
+			"logging_filter": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"default_behavior": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"KEEP", "DROP"}, false),
```
Comment (bflad) [inline aws/resource_aws_wafv2_web_acl_logging_configuration.go:50]: Nit: We can also use the constants here 👍 

```suggestion
							ValidateFunc: validation.StringInSlice(wafv2.FilterBehavior_Values(), false),
```

File: aws/resource_aws_wafv2_web_acl_logging_configuration.go [hunk 2]
```diff
@@ -37,6 +38,78 @@ func resourceAwsWafv2WebACLLoggingConfiguration() *schema.Resource {
...
+													Optional: true,
+													MaxItems: 1,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"label_name": {
+																Type:     schema.TypeString,
+																Required: true,
+																ValidateFunc: validation.All(
+																	validation.StringLenBetween(1, 1024),
+																	validation.StringMatch(regexp.MustCompile(`^[0-9A-Za-z_\-:]+$`), "must contain only alphanumeric characters, underscores, hyphens, and semicolons"),
```
Comment (bflad) [inline aws/resource_aws_wafv2_web_acl_logging_configuration.go:93]: Nit: Colons (`:`) not semicolons (`;`) 😎 

```suggestion
																	validation.StringMatch(regexp.MustCompile(`^[0-9A-Za-z_\-:]+$`), "must contain only alphanumeric characters, underscores, hyphens, and colons"),
```

File: aws/resource_aws_wafv2_web_acl_logging_configuration.go [hunk 3]
```diff
@@ -144,48 +225,195 @@ func resourceAwsWafv2WebACLLoggingConfigurationPut(d *schema.ResourceData, meta
...
 func resourceAwsWafv2WebACLLoggingConfigurationRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).wafv2conn
+
 	input := &wafv2.GetLoggingConfigurationInput{
 		ResourceArn: aws.String(d.Id()),
 	}
+
 	output, err := conn.GetLoggingConfiguration(input)
+
+	if !d.IsNewResource() && tfawserr.ErrCodeEquals(err, wafv2.ErrCodeWAFNonexistentItemException) {
```
Comment (bflad) [inline aws/resource_aws_wafv2_web_acl_logging_configuration.go:235]: ❤️

**Review by bflad — APPROVED** [review]
Two minor validation nits, otherwise looks great! 🚀 

Output from acceptance testing:

```
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_emptyRedactedFields (102.35s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_disappears_WebAcl (103.33s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_basic (104.64s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_disappears (121.55s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateSingleHeaderRedactedField (132.30s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateMethodRedactedField (135.95s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateUriPathRedactedField (137.04s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateEmptyRedactedFields (141.10s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateQueryStringRedactedField (141.97s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_updateMultipleRedactedFields (146.92s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_loggingFilter (176.18s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeResourceARNForceNew (188.33s)
--- PASS: TestAccAwsWafv2WebACLLoggingConfiguration_changeLogDestinationConfigsForceNew (193.05s)
```

File: aws/resource_aws_wafv2_web_acl_logging_configuration.go (hunk 2 above)
Comment (anGie44) [reply aws/resource_aws_wafv2_web_acl_logging_configuration.go:93]: ahhh 🙃 great catch!

**Pull request merged**